cmake/WlFunctions.cmake

```diff
@@ -79,6 +79,9 @@ macro(_common_compile_tasks)
     if(MINIZIP_STATIC_LIBRARIES)
       wl_include_directories(${NAME} ${MINIZIP_INCLUDE_DIRS})
       target_compile_options(${NAME} SYSTEM PUBLIC ${MINIZIP_CFLAGS})
+      if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        target_compile_options(${NAME} PUBLIC "-Wno-reserved-id-macro")
+      endif()
```

Comment:
Sorry, if I wasn't clear with my edit:
It should be sufficient to use the macros around the problematic includes (zip.h, unzip.h).
https://github.com/widelands/widelands/blob/efd890fd01556a957a0312b3c8a19e446f11a716/src/base/macros.h#L103-L104
Usage example:
https://github.com/widelands/widelands/blob/efd890fd01556a957a0312b3c8a19e446f11a716/src/base/i18n.cc#L49-L51

Reply:
> Sorry, if I wasn't clear with my edit:

No worries. You've been immensely helpful @matthiakl